bcipy/helpers/README.md

```diff
@@ -0,0 +1,140 @@
+```
+
+### Trigger Types
+
+The supported internal trigger types are as follows:
+
+```python
+NONTARGET = "nontarget"
+TARGET = "target"
+FIXATION = "fixation"
+PROMPT = "prompt"
+SYSTEM = "system"
+OFFSET = "offset"
+EVENT = "event"
+PREVIEW = "preview"
+```
+
+### TriggerHandler
+
+You can use the BciPy TriggerHandler to read and write any Triggers! 
+
+### Writing
+
+You will need three pieces of information to create a new handler for writing: 
+
+1. path where trigger data should be written
+2. name of the trigger file without extension
+3. a defined FlushFrequency. This sets how often the handler should write the data. Incrementally (FlushFrequency.EVERY) or at session end (FlushFrequency.END).
+
+```python
+from bcipy.helpers.triggers import TriggerHandler, FlushFrequency
+
+path_to_trigger_save_location = '.'
+trigger_file_name = 'triggers' # BciPy will add the correct extension. Currently, .txt is used.
+flush = FlushFrequency.END
+
+handler = TriggerHandler(path_to_trigger_save_location, trigger_file_name, flush)
+
+triggers = [
+    Trigger('test_trigger', TriggerType.SYSTEM, 1.0111),
+    Trigger('target', TriggerType.TARGET, 2),
+    Trigger('nontarget', TriggerType.NONTARGET, 3),
+]
+
+handler.add_triggers(triggers)
+handler.close() # this will call write one final time on any triggers added since last flush
+```
+
+
+### Loading
+
+To load a BciPy triggers.txt file, the TriggerHandler load method can be used. Because it is a staticmethod, the class does not need to be initialized before the load method is used. 
+
+```python
+from bcipy.helpers.triggers import TriggerHandler
+
+path_to_trigger_save_location = './triggers.txt'
+
+triggers = TriggerHandler.load(path_to_trigger_save_location)
+# it will load in data like this: 
+# [Trigger: label=[starting_offset] type=[offset] time=[-13149613.488788936], Trigger: label=[x] type=[prompt] time=[4.96745322458446]]
+```
+
+Alternately, you can pass offset and exclusion as keyword arguments to modify the behavior of `TriggerHandler.load`. Offset will add time to every timestamp loaded. Use this to correct any static system offsets! The default value for offset is 0.0. Exclusions can be used to pre-filter any unwanted Triggers, such as SYSTEM.
+
+```python
+from bcipy.helpers.triggers import TriggerHandler
+
+path_to_trigger_save_location = './triggers.txt'
+
+# exclude system triggers
+triggers = TriggerHandler.load(path_to_trigger_save_location, exclusion=[TriggerType.SYSTEM])
+
+# apply a 2 second offset to all timestamps loaded 
+triggers = TriggerHandler.load(path_to_trigger_save_location, offset=2.0)
+```
+
+### Trigger File Format
+
+Triggers as written from BciPy tasks are assumed to have the following structure,
+
+1. A single trigger is written per line with three columns (label type timestamp).
+2. If a clock offset is present, it is written with the type offset and will be added to any timestamp values written in the file. When using various clocks that don't 
```

Comment:
Is it possible to include multiple offsets? For example

```
starting_offset offset -5.0
another_offset offset 5.0
...
```

Reply:
That's a good callout, we only use the first offset written. The other will be loaded as a trigger and can be applied using other functions to the other loaded Triggers if desired. https://github.com/CAMBI-tech/BciPy/blob/1.5.1/bcipy/helpers/triggers.py#L274